tensorboard/components/tf_ng_tensorboard/types/api.ts

```diff
@@ -55,3 +55,19 @@ export interface Dashboard {
    */
   isReloadDisabled: boolean;
 }
+
+export interface PluginMetadata {
+  disable_reload: boolean;
+  enabled: boolean;
+  loading_mechanism: {
```

Comment:
This type isn’t quite right; it’s a discriminated union:

```typescript
type LoadingMechanism =
  | {type: 'IFRAME', module_path: string}
  | {type: 'CUSTOM_ELEMENT', element_name: string}
  | {type: 'NONE'}
  ;
```

Reply:
Ah you've already found the issue with it. I was going to address this in subsequent PR but decided to move that change over to this PR. Addressed!